resources/elasticache-memcacheclusters.go

```diff
@@ -62,21 +62,38 @@ func (l *ElasticacheCacheClusterLister) List(_ context.Context, o interface{}) (
 		}
 
 		resources = append(resources, &ElasticacheCacheCluster{
-			svc:       svc,
-			clusterID: cacheCluster.CacheClusterId,
-			status:    cacheCluster.CacheClusterStatus,
-			Tags:      tags.TagList,
+			svc:        svc,
+			clusterID:  cacheCluster.CacheClusterId,
+			status:     cacheCluster.CacheClusterStatus,
+			Tags:       tags.TagList,
+			serverless: false,
+		})
+	}
+
+	serverlessParams := &elasticache.DescribeServerlessCachesInput{MaxResults: aws.Int64(100)}
+	serverlessResp, serverlessErr := svc.DescribeServerlessCaches(serverlessParams)
+	if serverlessErr != nil {
+		return nil, serverlessErr
+	}
+
+	for _, serverlessCache := range serverlessResp.ServerlessCaches {
+		resources = append(resources, &ElasticacheCacheCluster{
+			svc:        svc,
+			clusterID:  serverlessCache.ServerlessCacheName,
+			status:     serverlessCache.Status,
+			serverless: true,
 		})
 	}
 
 	return resources, nil
 }
 
 type ElasticacheCacheCluster struct {
-	svc       elasticacheiface.ElastiCacheAPI
-	clusterID *string
-	status    *string
-	Tags      []*elasticache.Tag
+	svc        elasticacheiface.ElastiCacheAPI
+	clusterID  *string
+	status     *string
+	Tags       []*elasticache.Tag
+	serverless bool
 }
 
 func (i *ElasticacheCacheCluster) Properties() types.Properties {
```

Comment:
```suggestion
func (r *ElasticacheCacheCluster) Properties() types.Properties {
```

Reason: standardizing all pointers for resources to `r`